tarogarden/seedling.go

```diff
@@ -124,6 +128,38 @@ func (c Seedling) validateFields() error {
 	return nil
 }
 
+// validateGroupKey attempts to validate that the non-zero group key provided
+// with a seedling is owned by the daemon and can be used with this seedling.
+func (c Seedling) validateGroupKey(group asset.AssetGroup) error {
+	// We must be able to sign with the group key.
+	if !group.GroupKey.IsLocal() {
+		groupKeyBytes := c.GroupInfo.GroupPubKey.SerializeCompressed()
+		return fmt.Errorf("can't sign with group key %x", groupKeyBytes)
+	}
+
+	// The seedling asset type must match the group asset type.
+	if c.AssetType != group.Genesis.Type {
+		return fmt.Errorf("seedling type does not match "+
+			"group asset type %v", group.Genesis.Type)
+	}
+
+	return nil
+}
+
+// HasGroupKey checks if a seedling specifies a particular group key.
+func (c Seedling) HasGroupKey() bool {
+	// We have to nest these checks to avoid dereferencing a nil pointer and
```

Comment:
I'm not sure we have to, since Golang uses "short circuit" evaluation. For example if you have an `&&` operator and the first operand evaluates to `false`, then the value of the second operand doesn't change the outcome and is therefore skipped. So this could just be written as `if c.GroupInfo != nil && c.GroupInfo.GroupKey != nil {` without panicking.

The same goes for `||` operators, if the first evaluates to `true` by the way.

See https://stackoverflow.com/questions/53722288/short-circuit-evaluation-in-go